miscellaneous/metaDataRemover/tokenSorting.go

```diff
@@ -0,0 +1,200 @@
+	ret := make(map[string][]*interval)
+	for token, nonces := range tokens {
+		ret[token] = getIntervals(nonces)
+	}
+
+	return ret
+}
+
+func getIntervals(nonces []uint64) []*interval {
+	numNonces := len(nonces)
+	intervals := make([]*interval, 0)
+	for idx := 0; idx < numNonces; idx++ {
+		nonce := nonces[idx]
+		if idx+1 >= numNonces {
+			intervals = append(intervals, &interval{
+				start: nonce,
+				end:   nonce,
+			})
+			break
+		}
+
+		currInterval := &interval{start: nonce}
+		numConsecutiveNonces := uint64(0)
+		for idx < numNonces-1 {
+			currNonce := nonces[idx]
+			nextNonce := nonces[idx+1]
+			if nextNonce-currNonce > 1 {
+				break
+			}
+
+			numConsecutiveNonces++
+			idx++
+		}
+
+		currInterval.end = currInterval.start + numConsecutiveNonces
+		intervals = append(intervals, currInterval)
+	}
+
+	return intervals
+}
+
+func sortTokenIntervalsByMaxConsecutiveNonces(tokens map[string][]*interval) []*tokenWithInterval {
+	ret := make([]*tokenWithInterval, 0)
+	for token, intervals := range tokens {
+		for _, currInterval := range intervals {
+			ret = append(ret, &tokenWithInterval{
+				tokenID:  token,
+				interval: currInterval,
+			})
+
+		}
+	}
+
+	sort.SliceStable(ret, func(i, j int) bool {
+		consecutiveNonces1 := ret[i].interval.end - ret[i].interval.start + 1
+		consecutiveNonces2 := ret[j].interval.end - ret[j].interval.start + 1
+
+		if consecutiveNonces1 == consecutiveNonces2 {
+			return ret[i].tokenID < ret[j].tokenID
+		}
+
+		return consecutiveNonces1 > consecutiveNonces2
+	})
+
+	return ret
+}
+
+func groupTokenIntervalsInBulks(tokens []*tokenWithInterval, bulkSize uint64) [][]*tokenData {
+	bulks := make([][]*tokenData, 0, bulkSize)
+	currBulk := make(map[string][]*interval, 0)
+	numNoncesInBulk := uint64(0)
+
+	tokensCopy := make([]*tokenWithInterval, len(tokens))
+	copy(tokensCopy, tokens)
+
+	index := 0
+	for index < len(tokensCopy) {
+		currTokenData := tokensCopy[index]
+		currInterval := currTokenData.interval
+		currTokenID := currTokenData.tokenID
+
+		noncesInInterval := currInterval.end - currInterval.start + 1
+		availableSlots := bulkSize - numNoncesInBulk
+		if availableSlots >= noncesInInterval {
+			numNoncesInBulk += noncesInInterval
+			currBulk[currTokenID] = append(currBulk[currTokenID], currInterval)
+
+		} else {
+			first, second := splitInterval(currInterval, availableSlots)
+
+			numNoncesInBulk += availableSlots
+			currBulk[currTokenID] = append(currBulk[currTokenID], first)
+			tokensCopy = insert(tokensCopy, index+1, &tokenWithInterval{tokenID: currTokenID, interval: second})
+		}
+
+		bulkFull := numNoncesInBulk == bulkSize
+		lastInterval := index == len(tokensCopy)-1
+		if bulkFull || lastInterval {
+			bulks = append(bulks, tokensMapToOrderedArray(currBulk))
+
+			currBulk = make(map[string][]*interval, 0)
+			numNoncesInBulk = 0
+		}
+
+		index++
+	}
+
+	return bulks
+}
+
+func splitInterval(currInterval *interval, index uint64) (*interval, *interval) {
```

Comment:
these could have been part of the interval struct

Reply:
Agree, added this func as part of `interval` struct